src/par/FixColumnPartitions.c

```diff
@@ -99,18 +114,71 @@ int FixColumnPartitions_UpDownFaces(Mesh_ptr mesh, MRegion_ptr mr, MFace_ptr* up
     MSTK_Report("FixColumnPartitions","Mesh is not quite columnar, up and down faces do not align.",MSTK_FATAL);
   }
 
-  if (upcen[2] < dncen[2] - 1.e-12) {
-    rf = *up;
-    *up = *dn;
-    *dn = rf;
-    ret = 1;
-  } else if (upcen[2] > dncen[2] + 1.e-12) {
-    ret = 1;
-  } else {
-    /* find the first region upward that is proper */
-    MSTK_Report("FixColumnPartitions","Partitioning with degenerate cells is not yet supported.",MSTK_FATAL);
-    ret = 0;
+  if (up_unknown == 1) {
```

Comment:
Yes, it looks for a non-degenerate cell in either direction until it finds one. Using the outward normals is another option, but then it can be used regardless of whether the cell is degenerate and the top is known. If memory serves me right, that's how it's been originally. The caveat is that you have to be perfectly sure that the topology info is accurate, i.e. the direction of the normal is consistent with what face tells about it's orientation wrt to cell. I think at some point that wasn't the case for some of degenerate geometries.